src/browser/commands/assert-view/index.js

```diff
@@ -155,18 +155,43 @@ module.exports.default = browser => {
         testplaneCtx.assertViewResults.add({ stateName: state, refImg: refImg });
     };
 
-    session.addCommand("assertView", async function (state, selectors, opts = {}) {
+    const waitSelectorsForExist = async (browser, selectors) => {
         await Promise.map([].concat(selectors), async selector => {
-            await this.$(selector)
+            await browser
+                .$(selector)
                 .then(el => el.waitForExist())
                 .catch(() => {
                     throw new Error(
-                        `element ("${selector}") still not existing after ${this.options.waitforTimeout} ms`,
+                        `element ("${selector}") still not existing after ${browser.options.waitforTimeout} ms`,
                     );
                 });
         });
+    };
+
+    const assertViewBySelector = async (browser, state, selectors, opts) => {
+        await waitSelectorsForExist(browser, selectors);
 
         return assertView(state, selectors, opts);
+    };
+
+    const assertViewByViewport = async (state, opts) => {
+        opts = _.defaults(opts, {
+            allowViewportOverflow: true,
```

Comment:
if this option is not set, then an error is thrown when taking a screenshot of the viewport?

Reply:
yes